### PR TITLE
Setup Sentry

### PR DESCRIPTION
**Pull request by masonmcelvain**
Closes https://github.com/iFixit/checkout-ui-extensions/issues/24

Loosely follows Shopify's suggestions [here](https://shopify.dev/docs/api/checkout-ui-extensions/2024-01/error-handling#sentry).

## QA

- Uncaught errors in local dev should be reported to the [checkout-dev](https://ifixit.sentry.io/projects/checkout-dev) project in Sentry.
- The links on the thank you page should still work properly (see QA from https://github.com/iFixit/checkout-ui-extensions/pull/49).

### CODE REVIEW

**Comment by deltuh-vee** [comment]
dev_block ✊ 
Product page links aren't shown for line items in the checkout summary anymore.
CleanShot 2024-02-21 at 16 10 52@2x

**Comment by masonmcelvain** [comment]
Thanks @deltuh-vee! un_dev_block 👍🏻

**Comment by deltuh-vee** [comment]
QA 🎬 
* Uncaught errors in local dev are reported to checkout-dev in Sentry
* The links on the thank you page still work properly

File: extensions/ui-extension/src/models/product.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+               contexts: {
+                  graphql: {
+                     errors,
+                  },
+               },
+            });
+         }
+         const parsed = ProductSchema.safeParse(data);
+         if (!parsed.success) {
+            Sentry.captureMessage('Unexpected storefront API product shape', {
```
Comment (jarstelfox) [inline extensions/ui-extension/src/models/product.tsx:45]: TIL: Sentry has a nice wrapper for reporting just a string and wrapping it as a bare exception

https://github.com/getsentry/sentry-javascript/blob/36806f3f12e8000906ed56e520700299b26db17f/packages/core/src/scope.ts#L638

**Review by jarstelfox — APPROVED** [review]
CR 🌵

**Pull request merged**